### PR TITLE
Add python-cudnn-frontend

### PR DESCRIPTION
**Pull request by weiji14**
The cuDNN FrontEnd(FE) API is a C++ header-only library that wraps the cuDNN C backend API. Repo at https://github.com/NVIDIA/cudnn-frontend

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by weiji14** [comment]
@conda-forge/help-python-c, ready for review!

**Comment by xhochy** [comment]
If this is a header-only library, why does it need Python?

**Comment by weiji14** [comment]
This is built with the Python bindings (https://github.com/NVIDIA/cudnn-frontend?tab=readme-ov-file#python-api). I could try to remove `python` from the host and runtime deps and see if it works?

**Comment by xhochy** [comment]
It depends on what you want to achieve with this package. If it should be one for the headers, you should remove python. If you want to have the Python bindings, then this looks good.

**Comment by weiji14** [comment]
Yes, I'll be using this with the python bindings. Should I add a `-python` suffix perhaps?

**Comment by xhochy** [comment]
> Yes, I'll be using this with the python bindings. Should I add a `-python` suffix perhaps?

For these cases, we normally add a python- prefix.

File: recipes/cudnn-frontend/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  skip: true           # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - {{ stdlib("c") }}
+    - cmake >=3.17
+    - cudnn
```
Comment (xhochy) [inline recipes/cudnn-frontend/meta.yaml:27]: This should be in host and then you probably don't need to explicitly mention it in run anymore.

Comment (weiji14) [reply recipes/cudnn-frontend/meta.yaml]: Ok, done! Thanks again for checking :smiley:

**Comment by weiji14** [comment]
> In your conda_build_config.yaml, please change the name of `MACOSX_DEPLOYMENT_TARGET`, to `c_stdlib_version`!

Guessing that this is a false alarm since there's no `conda_build_config.yaml` file added? Can probably just ignore also since this recipe won't be built on macOS.

**Review by xhochy — APPROVED** [review]

**Pull request merged**